TsJournalTrello.gs

```diff
@@ -0,0 +1,361 @@
+   return config[2](args)
+    
+  } catch (error) {
+
+  var handleError = Assert.HandleError.DISPLAY_FULL
+
+  if (!PRODUCTION_VERSION_) {
+    handleError = Assert.HandleError.THROW
+  }
+
+  var assertConfig = {
+    error:          error,
+    userMessage:    config[1],
+    log:            Log_,
+    handleError:    handleError, 
+    sendErrorEmail: SEND_ERROR_EMAIL_, 
+    emailAddress:   ADMIN_EMAIL_ADDRESS_,
+    scriptName:     SCRIPT_NAME,
+    scriptVersion:  SCRIPT_VERSION,
+    }
+
+    Assert.handleError(assertConfig)
+    }
+    
+    } // eventHandler_()
+
+// Private event handlers
+// ----------------------
+
+/**
+ *
+ *
+ * @param {object} 
+ *
+ * @return {object}
+ */
+ 
+function linkHeaders_(DocId) {
+
+  Log_.functionEntryPoint()
+  
+  //Open the organisation spreadsheet
+  var orgData = SpreadsheetApp.openById(ORG_ID_)
+      .getSheetByName('Organisations')
+      .getDataRange()
+  
+  var data = orgData.getValues()
+  var timesheetUrl = null
+  var trelloBoardUrl = null
+  var trelloBoardId = null
+
+  // Get the organisation name from the journal name, text before '- journal' 
+  var str = ' - Journal'
+  var CHAR_TO_END_ = str.length
+  var orgName = DocumentApp.getActiveDocument().getName().slice(0, -CHAR_TO_END_)
+  var orgFound = false
+    
+  //Loop through the rows of data
+  orgFound = data.some(function(row)  {
+      
+    //check if the company name matches the orgName
+    var orgNameSearch = row[ORG_NAME_COL_]
+      
+    if (orgNameSearch === orgName) {
+    
+      //Get the timesheet and Trello Board Url
+      timesheetUrl = row[ORG_TS_COL_]
+      trelloBoardUrl = row[TRELLO_COL_]
+      return true
+      
+      }
+      
+  })   
+    
+  if (!orgFound) {
+    
+    Log_.warning('Organisation Name: ' + orgName + ' not found in Org Spreadsheet')
+    return
+      
+  } else {
+    
+    Log_.info('Organisation Name: ' + orgName + ' found in Org Spreadsheet')
+      
+  }
+   
+    
+  //Check the Trello Board Exists  
+  if (trelloBoardUrl === '') {
+    
+    Log_.warning('No Trello Board URL found in Org Spreadsheet')
+    return
+      
+  } else {
+      
+    //Get the Trello Board Id from the wedsite json return  
+    trelloBoardId = getTrelloBoardId_(trelloBoardUrl)
+    
+    if (trelloBoardId == null) {
+      Log_.warning('No new trello board Id found for ' + trelloBoardUrl)
+      return
+    }    
+    Log_.info('Trello Card Id: ' + trelloBoardId)
+        
+  } //if (trelloBoardUrl === '')
+      
+  //Open the Journal, loop through the paragraphs until the Heading 1 is found
+  //The next paragraph is the Trello Card Name, check it hasnt already been processed.
+  //If it hasnt, change the style to Heading 2 and add a Hyperlink to the Trello Card 
+  var journal = DocumentApp.openById(DocId)
+  var DocUrl = journal.getUrl()
+  var npBookmark = null
+ 
+  if (journal === null) { 
+  
+    Log_.warning('Invalid Journal ID' + DocId)
+   
+    return
+  } else {
```

Comment:
redundant else ... I'll let you pick up the later ones.

Reply:
done